MIP/mip-draft-FFS/README.md

```diff
@@ -0,0 +1,229 @@
+- equivocate (send a different vote to different validators or users)
+- vote :white_check_mark: for an invalid block 
+- vote :x: for a valid block
+
+### Main challenges
+
+To achieve crypto-economically secured Fast-Finality, we need to solve the following problems:
+
+1. design a _staking_ mechanism for the validators to stake assets, distribute rewards and manage slashing
+1. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for L2-finality
+1. _communicate_ the L2-finality status.
+
+In addition the following may require separate discussion, as it is a different procedure (postconfirmations are handled in smart contracts on L1, whereas L2-finality is handled off-chain)
+
+4. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for postconfirmation.
+
+
+### Components
+
+#### Staking (addresses 1. and 2.)
+
+The staking mechanism is implemented in a contract on L1.
+This contract provides the following functionalities:
+
+- join: a new validator can join the set of validators by staking some assets
+- exit: a validator exits and  get their stakes back
+- vote: receive a vote or a set of votes, verify the integrity of the votes (signatures) and the minimum threshold (e.g. 2/3)
+
+#### Generate an L2-finality certificate (addreses 3.)
+
+To ensure that the L2-finality status is made available to third-parties, we may publish our _proof_  (2/3 of attestations :white_check_mark:) to a data availability layer and get a _certificate_ that the proof is available.  
+This DA layer should offer a reliable _mempool_ for example as described [in this paper](https://arxiv.org/pdf/2105.11827).
+
+#### Handle postconfirmations (addresses 4.)
+
+The L1 contract will verify the L2-finality certificate. If the certificate is correct the block (or sequence of blocks) are _postconfirmed_. This requires handling who should send the certificate to the L1 contract, and how to verify the certificate.
+
+
+## Reference Implementation
+
+To simplify we assume that each validator stakes the same amount.
+The set of validators is in charge of validating a block at each round.
+
+A leader validator $V_l$ is elected in each round, and proposes the next transition (and block $B'$):  $B \xrightarrow{\ txs \ } B'$.
+They can do by sending a digest of $txs$ (Merkle root) and a digest of $B'$ (Merke root hash of $B'$), or a _change set_.
+Every other validator checks the validity of $B'$ and send their vote to $V_l$.
+Once $V_l$ has received more than 2/3 of :white_check_mark: for $B'$ they sent the votes (who voted :white_check_mark: and who voted :x:) to the L1 staking contract **and** to the DA.
+
+From there on two separate threads of event occur:
+
+- the DA layer returns an _availability certificate_ that a proof of 2/3 super-majority is available for block $B'$. This step should take O(1) second if we use a fast reliable mempool.
```

Comment:
A DA is nice, but I am not sure it is necessary and complicates the setup. Even if all honest nodes would drop at height H and have to recover from an earlier point H', they could do so and the guarantee to get to the same block root at height H is save. 

also not sure we should build our discussion around a reliable mempool. I assume this originates from the L1-mempool discussion, but for L2-finality it is not really necessary

Reply:
the purpose of the availability certificate is to ensure users can notice the post confirmation event. 
We can simplify this paragraph by focussing on the DA layer (a black box without mentioning a reliable mempool)?

Reply:
Yes that sounds good